### ISSUE TITLE
Issues with Magic RP on 1.20.1 on latest dev build

### ISSUE THREAD

**Issue by I0nTech** [elBukkit/MagicPlugin#1270]
With less than 6 spells in the spell hotbar, it fails to show the spell icons and shows this:
![image](https://github.com/elBukkit/MagicPlugin/assets/90792587/ba8f930c-c3f9-4e27-b178-b5ba2e0f983d)
This is the case on both this custom modified RP and the magic default RP. Both are correctly configured and display correctly with more than 6 spells in the hotbar. 

![image](https://github.com/elBukkit/MagicPlugin/assets/90792587/c23b8046-623e-4e5b-87f8-26fd647a0a18)

A fix by the community has been rolled out for 1.20+ packs, it'd be great if this change was posted to the magic repo.
https://discord.com/channels/580099743875727363/912845695256514621/1149750830560456725

**Comment by I0nTech** [comment]
EDIT: No, it does not seem like the community solution entirely works, the above issue when less than 6 spells are in the spell hotbar is still present. Seems more like a plugin-side issue, as the fonts themselves seem to be working correctly.

**Comment by NathanWolf** [comment]
<img width="388" alt="Screenshot 2023-09-11 at 2 03 38 PM" src="https://github.com/elBukkit/MagicPlugin/assets/522168/b2f32b0b-0697-4a82-8092-e548d20ddc29">

I'm not seeing this problem at all - is there anything else required to make it happen?

The community solution seems hacky at best- the hotbar can't just always use the magic font or it won't be able to show text.

**Comment by I0nTech** [comment]
> 

Not sure, what RP are you using? I'm on 1.20.1 paper with the latest magic dev build, with a modified magic RP - that might be the cause but I can't tell why.

**Comment by NathanWolf** [comment]
I am using the ModelEngine variant, though I think since ME integration is broken I can revert my server to use the default "all" RP. I will do that and test again, but I can't imagine it matters since they all share the same font files.

**Comment by I0nTech** [comment]
> 

Ah, let me test without modelengine installed. That might be the cause

**Comment by NathanWolf** [comment]
Updated my server to just use the basic "robes" RP and it seems fine, too.

I am not going to have time to debug a custom RP, sorry :(

Can you try it with the official RP?

**Comment by I0nTech** [comment]
> 

Sure, 1 second - I'll install default

**Comment by I0nTech** [comment]
![image](https://github.com/elBukkit/MagicPlugin/assets/90792587/b752c000-ca58-484a-84b6-52b499d8b496)
Yeah, same result

**Comment by I0nTech** [comment]
> ![image](https://user-images.githubusercontent.com/90792587/267149672-b752c000-ca58-484a-84b6-52b499d8b496.png) Yeah, same result

**Comment by NathanWolf** [comment]
It does seem like I should release a 1.20 RP with the correct pack-format version (15) but I can't imagine that fixing this problem.

Are you letting the server give you the RP or just adding it yourself to your RP folder?

**Comment by I0nTech** [comment]
> 

https://editor.elmakers.com/zcwq1oxy
Having magic send it over, can try putting it directly in the server config,yml file

Also you might want to hold off on fixing that MEG integration, MEG 4 is apparently a major overhaul of the system to use display entities instead of armor stands, and will likely break

**Comment by NathanWolf** [comment]
ME breaks their API constantly, I think they don't care anymore since getting in bed with MythicMobs. I'm not really interested in continuing to support integration with it. Which is unfortunate, it's an insanely cool plugin.

Can you try joining the `mine.elmakers.com` server and see if the RP works for you there?

**Comment by I0nTech** [comment]
> 

Which server are you on?

**Comment by NathanWolf** [comment]
`mine.elmakers.com`, it's just one server, many worlds- RP doesn't change between them.

**Comment by I0nTech** [comment]
> `mine.elmakers.com`, it's just one server, many worlds- RP doesn't change between them.

Can't tell as I don't have a wand on the survival server - can't navigate between servers either, for some reason

**Comment by NathanWolf** [comment]
Gave you a wand, weird that you did not spawn with the crafting ingredients though? Or a navigator?

**Comment by I0nTech** [comment]
Issue still seems to be present, even with the new RP:
![image](https://github.com/elBukkit/MagicPlugin/assets/90792587/8e1f8673-657d-4d75-8150-aa3faf583f12)

Going to remove all plugins but magic and see if that makes any difference...

**Comment by I0nTech** [comment]
Nope, issue is still present, using /getrp survival on the latest dev build you just posted, not sure if that's the updated RP or not.
![image](https://github.com/elBukkit/MagicPlugin/assets/90792587/74363ff0-d14e-4eb3-bc3c-0c2a36382692)

**Comment by NathanWolf** [comment]
You can `/getrp url` to see- the latest dev build should have 1.20 in the filename.

The only thing that changed is the pack-format, though- and historically that makes zero difference unless you try to download the RP and use it manually. The MC client won't let you use an outdated pack, even if it would work just fine if given by a server 🤷

**Comment by I0nTech** [comment]
> You can `/getrp url` to see- the latest dev build should have 1.20 in the filename.
> 
> The only thing that changed is the pack-format, though- and historically that makes zero difference unless you try to download the RP and use it manually. The MC client won't let you use an outdated pack, even if it would work just fine if given by a server 🤷

Uhhh, using this RP now, from the /getrp url link - issue is still present.
![image](https://github.com/elBukkit/MagicPlugin/assets/90792587/f7950e78-fc2e-4add-a81f-c0a55c795dda)

If you can join my server to see if it's my server or client or whatnot, that'd be very appreciated - no idea what's happening with this...

**Comment by I0nTech** [comment]
On 1.20.1, if that makes any difference - issue is present on a vanilla client too.

**Comment by NathanWolf** [comment]
If we move away from thinking this is some kind of version-specific problem then it could be a config issue on your end, nothing to do with the RP. Are you sure this is a new issue?

The default action_bar config looks like this- is it possible you're just missing the font config?

```
        action_bar:
          message: '$hotbar`{font:"default",text:" $extra"}`'
          font: "magic"
          open_message: ""
          interval: 250
          delay: 1500
```

**Comment by I0nTech** [comment]
> 

This is my GUI config, 
https://editor.elmakers.com/1q3zifjv

Specifically, I'm using gui_actionbar, and it works fine when there are more than 6 spells in the hotbar - less than that causes the issue shown. Can try and delete that file and try again, but all three versions of that message: '$hotbar`{font:"default",text:" $extra"}`' component yielded the same results.

**Comment by NathanWolf** [comment]
Well- let me try updating Paper on my server. Could be something changed or got bugged in a later version, I'm pretty far behind.

**Comment by NathanWolf** [comment]
Ah- ha.. broken on latest Paper.
Well... that sucks a bit!

**Comment by I0nTech** [comment]
> Well- let me try updating Paper on my server. Could be something changed or got bugged in a later version, I'm pretty far behind.

Yeah, even on magic default, no plugins, vanilla client 1.20.1 paper the issue's still there. Paper was the most likely culprit...

**Comment by NathanWolf** [comment]
Guess I'll be binary searching through the Paper build history until I find what broke it :|

**Comment by I0nTech** [comment]
> Guess I'll be binary searching through the Paper build history until I find what broke it :|

Oof. Wish you luck with your bogosort.

**Comment by NathanWolf** [comment]
Thanks :(
I didn't see anything suspicious in the Paper revisions so I'm guessing it came from some upstream merge.

**Comment by NathanWolf** [comment]
Suspicion confirmed, this got broken in Paper #126 which was just an upstream update.

So now I guess I gotta look at all the Spigot commits between Paper #104 (~2023-08-05) and #126 (~2023-08-13)

**Comment by NathanWolf** [comment]
Gosh this is a rabbit-hole.

I have a strong suspicion it was this change in the bungee-chat code:

https://github.com/SpigotMC/BungeeCord/commit/2a716bbc7f91887e31bbc37fb608581673893605

Which got pulled into Spigot here: https://hub.spigotmc.org/stash/projects/SPIGOT/repos/spigot/commits/3bce7935c844cdc6709ace44f2446455a87a8240#Bukkit-Patches/0010-BungeeCord-Chat-API.patch

Which would have made it into the Paper #126 update.

I can't easily prove this suspicion nor do I have idea how to fix or workaround whatever is broken.

**Comment by NathanWolf** [comment]
"Fixed" (worked around) in latest dev build and released as 10.8.10 (this seemed like a major enough issue for a release).

My suspicion about the problem was correct - the new bungee-chat API has a "fix" in it that prevents legacy text from using a font. It sets a "reset" flag on the chat component which has it ignore all prior directives, such as a font. But since there isn't a way to specify a font in the legacy chat format, I was kind of stuck.

I don't think I could argue against the fix even though it broke my specific use-case.

So instead I copied the previous version of the relevant legacy conversion code from the bungee-chat repo into Magic and I'm using it instead.
